### PR TITLE
[WIP] k8s rebase

### PR DESCRIPTION
**Pull request by soltysh**

### CODE REVIEW

**Comment by liggitt** [comment]
It begins

**Comment by kargakis** [comment]
dkjhx9l

**Comment by kargakis** [comment]
@soltysh you have included https://github.com/kubernetes/kubernetes/pull/14726 in the bump which is a fix for https://github.com/openshift/origin/issues/4785. We don't need https://github.com/openshift/origin/commit/62e7141e5542b6b7534ddd3d242c01cb19ff48da anymore so drop it.

@liggitt @deads2k please confirm

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -12,20 +12,24 @@
 		},
 		{
 			"ImportPath": "code.google.com/p/go-uuid/uuid",
-			"Comment": "null-12",
-			"Rev": "7dda39b2e7d5e265014674c5af696ba4186679e9"
+			"Comment": "null-15",
+			"Rev": "35bc42037350f0078e3c974c6ea690f1926603ab"
```
Comment (kargakis) [inline Godeps/Godeps.json]: I don't think this should get bumped. It's coming from docker/distribution.

**Comment by liggitt** [comment]
No, we actually want validation disabled, but we should override that in our wrapper to leave kubectl behavior unchanged

File: Godeps/Godeps.json [hunk 2]
```diff
@@ -781,27 +822,27 @@
...
 		{
 			"ImportPath": "github.com/shurcooL/sanitized_anchor_name",
-			"Rev": "244f5ac324cb97e1987ef901a0081a77bfd8e845"
+			"Rev": "9a8b7d4e8f347bfa230879db9d7d4e4d9e19f962"
 		},
 		{
 			"ImportPath": "github.com/skynetservices/skydns/backends/etcd",
-			"Comment": "2.5.2d",
-			"Rev": "bb2ebadc9746f23e4a296e3cbdb8c01e956baee1"
+			"Comment": "2.5.1a",
```
Comment (kargakis) [inline Godeps/Godeps.json]: I don't think we want skydns backleved

File: Godeps/Godeps.json [hunk 3]
```diff
@@ -843,12 +888,17 @@
...
 		{
-			"ImportPath": "github.com/vjeantet/asn1-ber",
-			"Rev": "85041cd0f4769ebf4a5ae600b1e921e630d6aff0"
+			"ImportPath": "github.com/vjeantet/goldap/message",
+			"Rev": "c4f081986193996a259a69268965dd16aca3d730"
 		},
 		{
 			"ImportPath": "github.com/vjeantet/ldapserver",
-			"Rev": "19fbc46ed12348d5122812c8303fb82e49b6c25d"
+			"Comment": "v1.0-10-g3a8d560",
```
Comment (kargakis) [inline Godeps/Godeps.json]: @liggitt you said you wanted to review latest changes to ldapserver, fyi

Comment (liggitt) [inline Godeps/Godeps.json]: Why did this change? It's not used upstream that I'm aware of? Seems like something weird happened with your godep save

File: Godeps/Godeps.json (hunk 2 above)
Comment (liggitt) [inline Godeps/Godeps.json]: Yeah… before we spend too much time with this, let's make sure the godep save was correct… this looks weird

**Comment by kargakis** [comment]
http://pastebin.com/CwgbnSYh

Lots of deps seem incorrectly touched

**Comment by soltysh** [comment]
Yeah, I've been fighting hard with godep yesterday. For some reason it was failing for me on every single step (restore and save) except for k8s part, which worked smoothly. I had to go get a lot of deps, though I don't remember any of those you guys mention being problematic, that's why it's weird for me.

**Comment by kargakis** [comment]
godep is a mess. Keep the Kube changes in a separate commit, and try to trim down any unnecessary bumps. Also check for any dropped patches in [go-restful](https://github.com/openshift/origin/commits/master/Godeps/_workspace/src/github.com/emicklei/go-restful) (you will pick these while cherry-picking), [skydns](https://github.com/openshift/origin/commits/master/Godeps/_workspace/src/github.com/skynetservices/skydns), [distribution](https://github.com/openshift/origin/commits/master/Godeps/_workspace/src/github.com/docker/distribution), [gophercloud](https://github.com/openshift/origin/tree/0e3685c8856799225113247392d8acf454e24083/Godeps/_workspace/src/github.com/rackspace/gophercloud). There may be others.

**Comment by soltysh** [comment]
I've fixed all the external deps. Next step I'll review k8s changes, because godep ignores by default _test.go files :cry: which we include everywhere.

**Comment by liggitt** [comment]
> godep ignores by default _test.go files

really? it's always pulled in _test.go files for me...

**Comment by soltysh** [comment]
@liggitt if you add -t flag it does, at least the latest version I have.

**Comment by soltysh** [comment]
I've split the shared libs bumps into separate commits, as @kargakis pointed me that.

**Comment by soltysh** [comment]
Superseded by #5143.

**Pull request closed without merging**